lib/request.js

```diff
@@ -319,7 +319,14 @@ module.exports = {
     var type = this.get('Content-Type');
     if (!type) return '';
 
-    return contentType.parse(type).parameters.charset || '';
+    try {
+      type = contentType.parse(type);
+    } catch (e) {
+      if (e instanceof TypeError) return '';
+      throw e
```

Comment:
Exception handling should always be specific. If a new error is thrown in the future then new error handling can be added here. +1 to keep as it is.